rllib/core/rl_module/rl_module.py

```diff
@@ -77,23 +100,120 @@ class RLModule(abc.ABC):
         More details here: https://github.com/pytorch/pytorch/issues/49726.
     """
 
-    @OverrideToImplementCustomLogic_CallToSuperRecommended
-    def __init__(self, config: Mapping[str, Any] = None) -> None:
-        self.config = config or {}
-        self.setup()
+    def __init_subclass__(cls, **kwargs):
```

Comment:
This defines a `__post_init__()` method for the class so that if there are things that we need to initialize that depend on some configs that are not set yet would end up here.

Reply:
Docstring please with an example and a description of which steps a Module goes through until it's fully initialized. This is very confusing right now.

Reply:
It is clarified in `__post_init__()` docstring. `__init_subclass__` is just adding `__post_init__` method to the subclass.